### PR TITLE
Add blog post 3.3 - 3.13 draft

### PR DESCRIPTION
**Pull request by arturi**

### CODE REVIEW

File: blog/2023-07-3.4-3.13.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+This is a big Uppy update post, covering relworkeases from [`3.3.0` to `3.13.0`](https://github.com/transloadit/uppy/compare/uppy%403.4.0...uppy%403.13.0)! 
+
+In this issue: the long-awaited conditional S3 Multipart, much improved performance when adding and validating 10k+ files, Server-sent events, assemblyOptions for Transloadit, stability improvements and bug fixes. 
+
+Please make a cup of something tasty ☕️ (in a non-spillable container), as this will be quite a ride.
+
+<!--truncate-->
+
+## Amazon S3
```
Comment (aduh95) [inline blog/2023-07-3.4-3.13.md:20]: I think we also want a specific section to address the deprecation of `@uppy/aws-s3-multipart` and the legacy `@uppy/aws-s3`.

```suggestion
## Amazon S3

### Merging of the plugins

After years of separating regular non-multipart uploads and multipart uploads, we are finally reuniting the two sibling plugins. As of now:

- `@uppy/aws-s3-multipart` is deprecated. You should use `@uppy/aws-s3` and set the `shouldUseMultipart: true` option to get the same behavior. We plan to keep it as an alias for the next major cycle and add a warning at the install when Uppy 4.0.0 will be released.
- `@uppy/aws-s3` now hosts two plugins:
  - the legacy one, untouched, to guarantee backward compatibility. We plan to get rid of it in the next major, we recommend moving away from it and report if you see something missing in the new plugin.
  - the new merged plugin, available if you pass `shouldUseMultipart` option.

Our plan for this merge is to maintain the backward compatibility until the next major, while also providing forward compatibility so you can try our new plugin and give us some feedback before we remove the old plugins for good.
```

File: blog/2023-07-3.4-3.13.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+1. Our users wanted to enable multipart conditionally per-file, for example opt in when file size exceeded 5 MB, not choose one or the other for the whole upload.
+2. The code in the original `@uppy/aws-s3` became more complex and error-prone, desperately needing a refactor. `@uppy/aws-s3-multipart` is more up-to-date, clean and organized, features better rate-limiting logic.
+
+To address these issues, we:
+
+1. Added the “regular” S3 functionality into the `@uppy/aws-s3-multipart`.
+2. Introduced a new option [`shouldUseMultipart: boolean | function`](https://uppy.io/docs/aws-s3-multipart/#shouldusemultipartfile) that can conditionally enable multipart uploads.
+
+```js
+uppy.use(AwsS3Multipart, {
```
Comment (aduh95) [inline blog/2023-07-3.4-3.13.md:37]:
```suggestion
uppy.use(AwsS3, {
```

File: blog/2023-07-3.4-3.13.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  shouldUseMultipart (file) {
+    // Use multipart only for files larger than 100MiB.
+    return file.size > 100 * 2 ** 20
+  },
+})
+```
+
+In the future major Uppy version we are planning to retire the original `@uppy/aws-s3`, making it an alias to `@uppy/aws-s3-multipart`. But the new name package name will be `@uppy/aws-s3` then. Not much is changing for the end users, besides this new `shouldUseMultipart` option.
```
Comment (Murderlon) [inline blog/2023-07-3.4-3.13.md:45]: This sentence is confusing to me. Let's make it actionable.

```suggestion
#### Migrate

The next major version will deprecate the `aws-s3-multipart` package. Until then you can use `shouldUseMultipart` in both plugins. However to get ahead of the deprecation, we advice using `aws-s3` already.
```

File: blog/2023-07-3.4-3.13.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+To address this issue and speed up the uploads by roughly 20%, we are introducing a new option: `getTemporarySecurityCredentials: boolean | function`. When `true`, both S3 and S3 Multipart uploads will be signed on the client, using the [AWS Security Federation Token](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetFederationToken.html), accessed via Companion and sent to the client, where it is cached and re-used.
+
+⚠️ You should opt-in into this only if you are comfortable giving end users direct access to signing files for your bucket.
+
+### Stability and bug fixes
+
+- We’ve improved the internal handling of `RateLimitedQueue` to make [retries more robust](https://github.com/transloadit/uppy/pull/4424) and increased [priority of `abort` and `complete`](https://github.com/transloadit/uppy/pull/4542) requests. 
+- While we were at it, we’ve also fixed [resuming in single-chunk uploads](https://github.com/transloadit/uppy/pull/4528), andimproved [chunk size calculation](https://github.com/transloadit/uppy/pull/4508).
+- [Golden Retriever integration with S3 Multipart](https://github.com/transloadit/uppy/pull/4526) was fixed.
```
Comment (Murderlon) [inline blog/2023-07-3.4-3.13.md:59]:
```suggestion
- While we were at it, we’ve also fixed [resuming in single-chunk uploads](https://github.com/transloadit/uppy/pull/4528), and improved [chunk size calculation](https://github.com/transloadit/uppy/pull/4508).
- [Golden Retriever integration with S3 Multipart](https://github.com/transloadit/uppy/pull/4526) was fixed.
```

File: blog/2023-07-3.4-3.13.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        caption: file.meta.caption,
+      },
+    }
+  }
+})
+```
+
+### Server-sent events
+
+Historically Transloadit supported progress updates via Socket.io. It is a robust and stable package, but came with a “price-tag” in the form of bundle size: [249.3 kB Minified](https://bundlephobia.com/package/socket.io@4.7.1). So it’s been on our minds for a while to replace it with something more lightweight, without breaking backwards-compatibility for the older clients (so simply removing Socket.io on the server in favor of plain WebSockets is not desireable).
```
Comment (Murderlon) [inline blog/2023-07-3.4-3.13.md:119]: You're checking out the server.

```suggestion
Historically Transloadit supported progress updates via Socket.io. It is a robust and stable package, but came with a “price-tag” in the form of bundle size: [38.3 kB Minified](https://bundlephobia.com/package/socket.io-client@4.7.1). So it’s been on our minds for a while to replace it with something more lightweight, without breaking backwards-compatibility for the older clients (so simply removing Socket.io on the server in favor of plain WebSockets is not desireable).
```

File: blog/2023-07-3.4-3.13.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+## Companion
+
+### Refresh tokens
+
+@uppy/companion: implement refresh for authentication tokens (#4448)
+
+### Bug fixes and improvements
+
+- Fixed 500 when file name contains non-ASCII chars
```
Comment (Murderlon) [inline blog/2023-07-3.4-3.13.md:139]: Not sure what case this is referring to

File: blog/2023-07-3.4-3.13.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+- Switched from aws-sdk v2 to @aws-sdk/* v3
+- Upgraded Grant dependency
+- We now send expire info for non-multipart uploads and added connection keep-alive to Dropbox
```
Comment (Murderlon) [inline blog/2023-07-3.4-3.13.md:142]: and what is the benefit outcome of that?

File: src/css/custom.css [hunk 8]
```diff
@@ -175,6 +175,10 @@ h2[itemprop='headline'] {
 	--ifm-h2-font-size: 1.75rem;
 }
 
+.markdown > h3 {
```
Comment (Murderlon) [inline src/css/custom.css:178]: I haven't checked it locally but we may not want this? In #118 I made sure to align everything consistently.

**Review by Murderlon — COMMENTED** [review]
Massive blog post! Mostly looking good, some changes needed

File: blog/2023-07-3.4-3.13.md (hunk 5 above)
Comment (arturi) [reply blog/2023-07-3.4-3.13.md:132]: Ohh haha, I thought it was a bit much :)) Sorry, thanks for catching!

File: blog/2023-07-3.4-3.13.md (hunk 6 above)
Comment (arturi) [reply blog/2023-07-3.4-3.13.md:152]: https://github.com/transloadit/uppy/pull/4493

Comment (arturi) [reply blog/2023-07-3.4-3.13.md:152]: I'll rephrase.

File: src/css/custom.css (hunk 8 above)
Comment (arturi) [reply src/css/custom.css:178]: **Before: h2 is almost indistinguishable from h3 to my eye:**
Screenshot from 2023-07-28 13-36-33

**After:**
Screenshot from 2023-07-28 13-36-21

File: blog/2023-07-3.4-3.13.md (hunk 7 above)
Comment (arturi) [reply blog/2023-07-3.4-3.13.md:155]: Added “connection stability”, @aduh95 is there more to share here, reason for sending expire for non-multipart? It was supposed to be both plugins I see, one was just forgotten.

**Review by aduh95 — APPROVED** [review]

**Pull request merged**